src/main/java/uk/ac/soton/itinnovation/security/systemmodeller/rest/ModelController.java

```diff
@@ -1662,18 +1664,39 @@ public ResponseEntity<JobResponseDTO> calculateRecommendations(
         return ResponseEntity.accepted().headers(headers).body(response);
     }
 
-    @GetMapping("/models/{modelId}/recommendations/status/{jobId}")
-    public ResponseEntity<RecStatus> checkRecJobStatus(
+    @PostMapping("/models/{modelId}/recommendations/{jobId}/cancel")
+    public ResponseEntity<RecStatus> cancelRecJob(
+            @PathVariable String modelId, @PathVariable String jobId) {
+
+        logger.info("Got request to cancel recommendation task for model: {}, jobId: {}", modelId, jobId);
+
+		synchronized(this) {
+			final Model model = secureUrlHelper.getModelFromUrlThrowingException(modelId, WebKeyRole.WRITE);
+			Progress progress = modelObjectsHelper.getTaskProgressOfModel(RECOMMENDATIONS, model);
+			progress.setMessage("Cancelling");
+			recommendationsService.updateRecStatus(jobId, RecStatus.ABORTED);
+		}
+
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @GetMapping("/models/{modelId}/recommendations/{jobId}/status")
+    public ResponseEntity<JobResponseDTO> checkRecJobStatust(
             @PathVariable String modelId, @PathVariable String jobId) {
 
         logger.info("Got request for jobId {} status", jobId);
 
-        return recommendationsService.getRecStatus(jobId)
-                .map(ResponseEntity::ok)
-                .orElse(ResponseEntity.notFound().build());
+        recommendationsService.updateRecStatus(jobId, RecStatus.ABORTED);
```

Comment:
This line looks suspicious for simply checking the status of a job. Probably added by accident, as it was also added in the cancelRecJob method.